sled-agent/src/rack_setup/service.rs

```diff
@@ -724,17 +728,20 @@ impl ServiceInner {
 
     async fn wait_for_timesync(
         &self,
-        sled_addresses: &Vec<SocketAddrV6>,
+        ntp_admin_addresses: &Vec<SocketAddrV6>,
```

Comment:
I realize this is consistent with what was already here, but: by accepting a list of socket addrs, we end up rebuilding an `NtpAdminClient` for every sled inside every attempt of the `retry_notify` loop. Maybe we should take a `&[NtpAdminClient]` instead of a list of socket addrs, so we only create the clients once? (I have a vague memory of reqwest clients being relatively expensive to construct, but maybe we solved that some other way?)